### PR TITLE
Add lxsocdoc

### PR DESCRIPTION
**Pull request by xobs**
As discussed in #367 #367

### CODE REVIEW

File: litex/soc/doc/__init__.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+def sub_csr_bit_range(busword, csr, offset):
+    nwords = (csr.size + busword - 1)//busword
+    i = nwords - offset - 1
+    nbits = min(csr.size - i*busword, busword) - 1
+    name = (csr.name + str(i) if nwords > 1 else csr.name).upper()
+    origin = i*busword
+    return (origin, nbits, name)
+
+
+def print_svd_register(csr, csr_address, description, length, svd):
```
Comment (mithro) [inline litex/soc/doc/__init__.py]: The `svd` stuff should probably be in the same place we generate the json and c headers rather than in the doc?

Comment (xobs) [reply litex/soc/doc/__init__.py]: Is there an easy way in python to convert a print() to a string concatenation?  The existing litex stuff creates one big string inside the functions in `export.py`, then writes them all out at once.  lxsocdoc, meanwhile, takes a stream-based approach and generally writes to the files as it goes.

Comment (enjoy-digital) [reply litex/soc/doc/__init__.py]: Maybe something similar to what we are doing in the build backends:
````
from litex.build import tools
svd = []
svd.append("line1")
svd.append("line2")
tools.write_to_file("registers.svd", "\n".join(svd))
````

Comment (xobs) [reply litex/soc/doc/__init__.py]: As per your suggestion, it's now in `export.py`.

Comment (enjoy-digital) [reply litex/soc/doc/__init__.py]: Thanks!

**Comment by enjoy-digital** [comment]
Thanks @xobs, this seems fine for an initial integration. Do you have some code snippet demonstrating how to use it on a existing SoC?

**Comment by xobs** [comment]
We do this on Fomu: https://github.com/im-tomu/foboot/blob/master/hw/foboot-bitstream.py#L449-L452

```python
    vns = builder.build()
    soc.do_exit(vns)
    lxsocdoc.generate_docs(soc, "build/documentation/", project_name="Fomu Bootloader", author="Sean Cross")
    lxsocdoc.generate_svd(soc, "build/software", vendor="Foosn", name="Fomu")
```

There are instructions inside the lxsocdoc [README file](https://github.com/litex-hub/lxsocdoc#usage), however those obviously will be lost with the merge into litex.

**Comment by enjoy-digital** [comment]
I'm merging and will add the README file of lxsocdoc in `litex/doc` as `lxsocdoc.md`, thanks.

**Pull request merged**